lib/truffle/truffle/cext.rb

```diff
@@ -120,8 +116,12 @@ def size
       Truffle::CExt.string_pointer_size(@string)
     end
 
-    def to_native
-      Truffle::FFI::Pointer.new(Truffle::CExt.string_pointer_to_native(@string))
+    def pointer?
+      true
+    end
+
+    def address
+      @address ||= Truffle::CExt.string_pointer_to_native(@string)
```

Comment:
Why storing the address as a field here?

Reply:
It can't change, so I thought caching it would be nice, especially if reading over and over. But I can back it out if you're concerned.

Reply:
It means 2 different Shapes for RStringPtr.
We need some kind of polymorphism here anyway (already native or not), but it's already expressed in `string_pointer_to_native`.
But it does avoid going through the Rope, Java Pointer and get the address everytime.
Both are fine.